### PR TITLE
Switch logging to Microsoft.Extensions.Logging in preparation for .NE…

### PR DESCRIPTION
**Pull request by drichelson**
…T core conversion.

I spun up the hello world client and specified a console logger factory and it works!

### CODE REVIEW

File: src/LaunchDarkly.Client/Clause.cs [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace LaunchDarkly.Client
 {
     class Clause
     {
-        private static readonly ILog Logger = LogProvider.For<Clause>();
+        private readonly ILogger Logger = LdLogger.CreateLogger<Clause>();
```
Comment (jkodumal) [inline src/LaunchDarkly.Client/Clause.cs]: Not static?

File: src/LaunchDarkly.Client/LaunchDarkly.Client.csproj [hunk 2]
```diff
@@ -37,6 +37,18 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
+    <Reference Include="Microsoft.Extensions.DependencyInjection.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>\\Mac\Home\Documents\packages\Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0\lib\netstandard1.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll</HintPath>
```
Comment (jkodumal) [inline src/LaunchDarkly.Client/LaunchDarkly.Client.csproj]: What are the implications of these hardcoded paths (which probably won't resolve for anyone else)?

Comment (drichelson) [inline src/LaunchDarkly.Client/LaunchDarkly.Client.csproj]: That explains the failing build

File: src/LaunchDarkly.Client/Clause.cs (hunk 1 above)
Comment (drichelson) [inline src/LaunchDarkly.Client/Clause.cs]: fixed

File: src/LaunchDarkly.Client/LaunchDarkly.Client.csproj (hunk 2 above)
Comment (drichelson) [inline src/LaunchDarkly.Client/LaunchDarkly.Client.csproj]: When I added these dependencies Visual Studio created this hint path, but we want them to be the relative paths within the repo that NuGet creates... I changed it and the AppVeyor build now passes

**Pull request merged**